### PR TITLE
[misc] Fixed duplicate saving incoming requests in the final clause if request was successful

### PR DESCRIPTION
**Pull request by veronikaslc**

### CODE REVIEW

File: core/server/src/main/java/org/phenotips/remote/server/internal/IncomingSearchRequestProcessor.java [hunk 1]
```diff
@@ -96,16 +97,19 @@ public JSONObject processHTTPSearchRequest(ApiDataConverter apiVersionSpecificCo
...
+            // save raw request data for audit purposes only
+            this.saveUnprocessedRequest(stringJson, remoteServerId, apiVersionSpecificConverter.getApiVersion());
             return apiVersionSpecificConverter.generateWrongInputDataResponse(ex.getMessage());
         } catch (Exception ex) {
             this.logger.error("CODE Error: {}", ex);
-            return apiVersionSpecificConverter.generateInternalServerErrorResponse(null);
-        } finally {
             // save raw request data for audit purposes only
             this.saveUnprocessedRequest(stringJson, remoteServerId, apiVersionSpecificConverter.getApiVersion());
+            return apiVersionSpecificConverter.generateInternalServerErrorResponse(null);
```
Comment (allasm) [inline core/server/src/main/java/org/phenotips/remote/server/internal/IncomingSearchRequestProcessor.java]: Hmm, there must be a solution with less code duplication. For example, a "requestIncomplete" variable, initially initialised to "true", and set to false right before the last return statement in the normal code flow. Then in `finally` we do `if (requestIncomplete) { .... }`

Comment (allasm) [inline core/server/src/main/java/org/phenotips/remote/server/internal/IncomingSearchRequestProcessor.java]: Hmm, there must be a solution with less code duplication. For example, a "requestIncomplete" variable, initially initialised to "true", and set to false right before the last return statement in the normal code flow. Then in `finally` we do `if (requestIncomplete) { .... }`

**Comment by veronikaslc** [comment]
@allasm comments addressed

**Comment by veronikaslc** [comment]
@allasm commits squashed

**Comment by allasm** [comment]
Code approved, to be merged after testing

**Comment by allasm** [comment]
Closing this in favour of tested commits in the `stable-1.1.x` branch that have to be cherry-picked in the correct order into `master`

**Pull request closed without merging**